### ISSUE TITLE
UnusedImports false positive for annotation used in another annotation attributes (with type resolution)

### ISSUE THREAD

**Issue by vladimirfx** [detekt/detekt#3246]
## Expected Behavior
When an annotation used in another annotation attributes that annotation import should not marked as unused.

## Steps to Reproduce

```kotlin
import somepkg.AnnotationA
import somepkg.AnnotationB

class SomeClass {
    @AnnotationB(attribute = AnnotationA)
    val someProp: Type
}

```

In such code `import somepkg.AnnotationA` is reported as unused.
## Your Environment
* Version of detekt used: 1.15.0-RC1
* Version of Gradle used (if applicable): 6.7.1

**Comment by cortinico** [comment]
Have you enabled type resolution? This rule has a mixed behavior (see #2994) based on wether you have Type Resolution enabled or not: https://detekt.github.io/detekt/type-resolution.html

I'm adding a test with your snippet and it seems to work fine:
https://github.com/detekt/detekt/pull/3255

**Comment by vladimirfx** [comment]
Yes I’ve use type resolution. Hm it seems I missing some important detail...
 In your test case there is no attribute in AnnotationB. How it compiles?

**Comment by cortinico** [comment]
> Hm it seems I missing some important detail...
> In your test case there is no attribute in AnnotationB. How it compiles?

Yup you're right. I've updated the test with a correct version. Unfortunately we don't have support for compiling snippets when there is more than one file involved.

To be fair, the snippet you posted initially was also invalid. You have to instantiate an annotation when passing it as parameter to another annotation: `@AnnotationB(attribute = AnnotationA())`.
Would you be able to provide a different snippet where we can reproduce?

**Comment by schalkms** [comment]
Closing this, since tests have been added for the initial code snippet. If new false-positives for this rule appear, please feel free to open a new issue. Thanks!
